data_steward/retraction/retract_data_bq.py

```diff
@@ -389,24 +408,40 @@ def skip_dataset_retraction(dataset_id, retraction_type) -> bool:
     msg_no_ehr = (
         f'because {RETRACTION_ONLY_EHR} is specified. This is a {get_dataset_type(dataset_id)}'
         ' dataset and it does not contain data from EHR.')
-    msg_no_mapping = (
-        f'because the script could not find mapping info for identifying which data came from EHR.'
-    )
 
     if retraction_type == RETRACTION_ONLY_EHR:
         if is_rdr_dataset(dataset_id) or is_fitbit_dataset(dataset_id):
             LOGGER.warning(f'{msg_skip}{msg_no_ehr}')
             return True
 
-        # TODO Remove this if statement when this script is ready to handle OTHER and sandbox datasets.
-        if get_dataset_type(dataset_id) == OTHER or is_sandbox_dataset(
-                dataset_id):
-            LOGGER.warning(f'{msg_skip}{msg_no_mapping}')
-            return True
-
     return False
 
 
+def get_primary_key_for_sandbox_table(table: str) -> str:
+    """
+    NOTE This is intended for `only_ehr` retraction for sandbox datasets.
+    
+    When retracting sandbox tables for `only_ehr`, map/ext tables do not
+    exist for them. We must look at its primary key instead, in addition to person_id.
+    
+    This function looks at the sandbox table name, gets its domain, and returns the key
+    we must use for retraction. 
+
+    :param table: name of the sandbox table that needs retraction
+    :return: column name that the retraction must run on. 
+        Returns '' if the domain table cannot be identified, the domain table
+        does not contain person_id, or the domain table does not contain EHR data.
+    """
+    domain_tables = set(AOU_REQUIRED) - set(NON_PID_TABLES)
+
+    for domain in domain_tables:
+        if domain in table:
+            if domain == DEATH:
+                return f"{PERSON}_id"
+            return f"{domain}_id"
```

Comment:
This is tricky, you may get observation_id for observation_period. I would suggest using 'endswith' and making sure the domain_id exists in the table, along with person_id.

Reply:
I assumed we do not have to retract `observation_period` for `only_ehr` since it is not `AOU_REQUIRED`. But I was wrong. Some HPO sites send `observation_period` records. Other non-`AOU_REQUIRED` tables, such as `drug_era`, also have EHR data. Let me update the script.
I will also add a check if the domain_id exists in the table.

Thanks for pointing it out, @nishanthpp93!